sdk/monitor/azingest/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,76 @@
+# Troubleshooting Azure Monitor Ingestion module
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Monitor Ingestion module for Go.
+
+## Table of contents
+
+* [General troubleshooting](#general-troubleshooting)
+    * [Error Handling](#error-handling)
+    * [Logging](#logging)
+* [Troubleshooting logs ingestion](#troubleshooting-logs-ingestion)
+    * [Troubleshooting authorization errors](#troubleshooting-authorization-errors)
+    * [Troubleshooting missing logs](#troubleshooting-missing-logs)
+    * [Troubleshooting slow logs upload](#troubleshooting-slow-logs-upload)
+
+## General troubleshooting
+
+### Error Handling
+
+All methods which send HTTP requests return `*azcore.ResponseError` when these requests fail. `ResponseError` has error details and the raw response from Monitor Ingestion.
+
+### Logging
+
+To troubleshoot issues with the module, first enable logging to monitor the behavior of the application. The errors and warnings in the logs generally provide useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+This module uses the logging implementation in `azcore`. To turn on logging for all Azure SDK modules, set `AZURE_SDK_GO_LOGGING` to `all`. By default, the logger writes to stderr. Use the `azcore/log` package to control log output. For example, logging only HTTP request and response events, and printing them to stdout:
+
+```go
+import azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+
+// Print log events to stdout
+azlog.SetListener(func(cls azlog.Event, msg string) {
+	fmt.Println(msg)
+})
+
+// Includes only requests and responses in credential logs
+azlog.SetEvents(azlog.EventRequest, azlog.EventResponse)
+```
+
+## Troubleshooting logs ingestion
+
+### Troubleshooting authorization errors
+
+If you get an error with HTTP status code 403 (Forbidden), it means that the provided credentials have insufficient permissions to upload logs to the specified Data Collection Endpoint (DCE) and Data Collection Rule (DCR) ID.
+
+```text
+"error": {
+"code": "OperationFailed",
+"message": "The authentication token provided does not have access to ingest data for the data collection rule with immutable Id '***' PipelineAccessResult: AccessGranted: False, IsRbacPresent: False, IsDcrDceBindingValid: , DcrArmId: /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr, Message: Required autorization action was not found for tenantId *** objectId *** on resourceId /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr ConfigurationId: ***.."
```

Comment:
```suggestion
"message": "The authentication token provided does not have access to ingest data for the data collection rule with immutable Id '***' PipelineAccessResult: AccessGranted: False, IsRbacPresent: False, IsDcrDceBindingValid: , DcrArmId: /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr, Message: Required authorization action was not found for tenantId *** objectId *** on resourceId /subscriptions/***/resourceGroups/***/providers/Microsoft.Insights/dataCollectionRules/az-dcr ConfigurationId: ***.."
```
Hopefully that type is just in this document.  If not please make a bug and fix the SDK or worse if the message comes from the service file a bug on the service.

Reply:
Following up to check if a bug